test/extensions/common/dynamic_forward_proxy/dns_cache_impl_test.cc

```diff
@@ -1006,6 +1024,98 @@ TEST(UtilityTest, PrepareDnsRefreshStrategy) {
   }
 }
 
+TEST_F(DnsCacheImplTest, ResolveSuccessWithCaching) {
+  // Configure the cache.
+  MockKeyValueStoreFactory factory;
+  EXPECT_CALL(factory, createEmptyConfigProto()).WillRepeatedly(Invoke([]() {
+    return std::make_unique<
+        envoy::extensions::key_value::file_based::v3::FileBasedKeyValueStoreConfig>();
+  }));
+  MockKeyValueStore* store{};
+  EXPECT_CALL(factory, createStore(_, _, _, _)).WillOnce(Invoke([&store]() {
+    auto ret = std::make_unique<NiceMock<MockKeyValueStore>>();
+    store = ret.get();
+    // Make sure there's an attempt to load from the key value store.
+    EXPECT_CALL(*store, iterate);
+    return ret;
+  }));
```

Comment:
Should this be implemented on the MockKeyValueStoreFactory itself? Having them in the test case here is a bit distracting from what the test is actually testing

Reply:
maybe?  Honestly I regularly get bit by expectations in mock timer where I don't understand what's going on, so I'm hesitant to add complexity like that to the mock base classes

Reply:
Sure not a big deal, just what I usually see done with mocks. Agree that MockTimer is especially confusing :)